server/resp/src/main/java/org/infinispan/server/resp/commands/cluster/SegmentSlotRelation.java

```diff
@@ -0,0 +1,57 @@
+package org.infinispan.server.resp.commands.cluster;
+
+import static org.infinispan.commons.logging.Log.CONFIG;
+
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.infinispan.distribution.ch.impl.CRC16HashFunctionPartitioner;
+
+public class SegmentSlotRelation {
+
+   static final int SLOT_SIZE = 1 << 14;
+   private final int segmentSize;
+   private final int width;
+   private final Map<Integer, List<Integer>> relation;
```

Comment:
I am not sure I understand the purpose of this Map. It seems to just be wasting memory. The way it is implemented here it is very simple to map a segment to a slot and vice versa with just simple math operations with just the segment count.